script.js

```diff
@@ -1,4 +1,102 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// Declare default output value as global variable
+var myOutputValue = 'hello world';
+// Declare variable tracking the number of attempts
+var attemptsCount = 0;
+// Declare variable tracking the number of wins by the user
+var winCount_user = 0;
+// Declare variable tracking the number of wins by the computer
+var winCount_com = 0;
+// Declare variable tracking the number of draws
+var drawCount = 0;
+// Declare 3 variables containing the string of 3 possible options
+var STONE = 'stone';
+var SCISSORS = 'scissors';
+var PAPER = 'paper';
+// Declare an array of the game options
+var gameOptions = [STONE, SCISSORS, PAPER];
+
+
+
+
+// Define a function that randomly generates either scissors, paper or stone
+var randomDraw = function () {
+  console.log('running randomDraw function');
+  
+  // Generate a random decimal between 0 to 3, inclusive of 0 and excluding 3
+  var randomDecimal = Math.random() * 3;
+  // Remove decimal to generate a random integer between 0 to 2, inclusive of 0 and 2
+  var randomInteger = Math.floor(randomDecimal);
+  // Assign string value of the array 'gameOptions' indexed by the random integer to the variable randomOutput
+  var randomOutput = gameOptions[randomInteger];
+
+  console.log('randomOutput');
+  console.log(randomOutput);
+  // Return the randomly chosen option, scissors paper or stone
+  return randomOutput;
+};
+
+
+
+// Define a function that checks if the user input is valid i.e. scissors, paper or stone
+var checkInput = function (str) {
+  // Convert input string characters to lowercase
+  var input = str.toLowerCase();
+  console.log('Running user input validation');
+  if (input != gameOptions[0] && input != gameOptions[1] && input != gameOptions[2]) {
+    return false;
+  }
+  return true;
+}
```

Comment:
Good job checking input as a function. 

If you like you could try using the includes method to check if the input has scissors paper or stone 

on line 45 it'll be something like `if (gameOptions.includes(input))
` this will return false if there isn't any scissors, paper or stone

Read more about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

Reply:
thanks for the suggestion! I think that really helps reduce the amount of code :D